server/app/repository/ProgramRepository.java

```diff
@@ -173,6 +173,32 @@ public CompletableFuture<Program> getActiveProgramFromSlug(String slug) {
         executionContext.current());
   }
 
+  /** Get the current draft program with the provided slug. */
+  public CompletableFuture<Program> getDraftProgramFromSlug(String slug) {
+    return supplyAsync(
+        () -> {
+          for (Program program : database.find(Program.class).where().isNull("slug").findList()) {
+            program.getSlug();
+            program.save();
+          }
+
+          Optional<Version> version = versionRepository.get().getDraftVersion();
+
+          if (version.isEmpty()) {
+            throw new RuntimeException(new ProgramNotFoundException(slug));
+          }
+
+          ImmutableList<Program> draftPrograms = version.get().getPrograms();
+          List<Program> programsMatchingSlug =
+              database.find(Program.class).where().eq("slug", slug).findList();
```

Comment:
**suggestion (blocking):** I don't think this second query is necessary? Looks like you could loop through `draftPrograms` and find the one with the matching `slug`